### PR TITLE
Ivy-bar-chart-summary-and-barChart-template

### PR DESCRIPTION
**Pull request by 20chen-7**
# Description
Please include the exact bug/functionality description and a summary of the changes/ related issues. Please also include any other relevant motivation and context:

## Main changes explained:
- Rename barChart.js for template…
- Create HoursCompletedBarChart.js for introducing new components …
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. go to `http://localhost:3000/totalorgsummary`, to check the Hours Completed bar, the start date and end date, last start date and last end date will replaced by date filter later. Here is just an example.

## Screenshots or videos of changes:
Screenshot 2024-08-03 at 1 26 09 PM

Include the information the reviewers need to know.

### CODE REVIEW

File: src/components/TotalOrgSummary/TotalOrgSummary.jsx [hunk 1]
```diff
@@ -1,29 +1,49 @@
...
 import SkeletonLoading from '../common/SkeletonLoading';
 import '../Header/DarkMode.css';
 import './TotalOrgSummary.css';
 
 // components
 import VolunteerHoursDistribution from './VolunteerHoursDistribution/VolunteerHoursDistribution';
 import AccordianWrapper from './AccordianWrapper/AccordianWrapper';
+import HoursCompletedBarChart from './HoursCompleted/HoursCompletedBarChart';
 
 const startDate = '2016-01-01';
```
Comment (cgomezhub) [inline src/components/TotalOrgSummary/TotalOrgSummary.jsx:17]: you can try this   code to compare the last full week with the previous full week:


function calculateFromDate() {
  const currentDate = new Date();
  currentDate.setHours(0, 0, 0, 0);
  const dayOfWeek = currentDate.getDay();
  const daysToSubtract = dayOfWeek === 0 ? 0 : dayOfWeek;
  currentDate.setDate(currentDate.getDate() - daysToSubtract - 7);
  return currentDate.toISOString().split('T')[0];
}

function calculateToDate() {
  const currentDate = new Date();
  currentDate.setHours(0, 0, 0, 0);
  const dayOfWeek = currentDate.getDay();
  const daysToAdd = dayOfWeek === 6 ? 0 : -1 - dayOfWeek;
  currentDate.setDate(currentDate.getDate() + daysToAdd);
  return currentDate.toISOString().split('T')[0];
}

function calculateFromOverDate() {
  const currentDate = new Date();
  currentDate.setHours(0, 0, 0, 0);
  const dayOfWeek = currentDate.getDay();
  const daysToSubtract = dayOfWeek === 0 ? 0 : dayOfWeek;
  currentDate.setDate(currentDate.getDate() - daysToSubtract - 14);
  return currentDate.toISOString().split('T')[0];
}

function calculateToOverDate() {
  const currentDate = new Date();
  currentDate.setHours(0, 0, 0, 0);
  const dayOfWeek = currentDate.getDay();
  const daysToAdd = dayOfWeek === 6 ? 0 : -8 - dayOfWeek;
  currentDate.setDate(currentDate.getDate() + daysToAdd);
  return currentDate.toISOString().split('T')[0];
}

const startDate = calculateFromDate();
const endDate = calculateToDate();
const lastStartDate = calculateFromOverDate();
const lastEndDate = calculateToOverDate();

**Comment by jayyy-s** [comment]
Tested and the bar charts look fine as in the screenshot.
PR 2508 sc

**Review by SnelPatare — DISMISSED** [review]
Changes working well. I have one suggestion 
Light mode looks good.
In dark mode, it is difficult to read black letters. It will be great if color can be changed to light one in dark mode. 
Please find images below
image

image

**Review by oliviahyw — DISMISSED** [review]
I have tested this PR and it's working as expected.
PR2508

**Comment by FayeLV** [comment]
Hi,
Great job! Everything looks good overall, but there are a couple of issues that appear when the screen size is between 995 and 1184 pixels:
Bar Chart Labels Overlapping: The value labels on top of the bars (e.g., 122.97 and 0%) seem to overlap with each other, which can make the data hard to read.
Y-Axis Alignment Issue: It looks like the bottom of the Y-axis and the bottom of the bars are not properly aligned. The "0" tick on the Y-axis is slightly offset from the base of the bars, making the chart look a bit cluttered and unclear.
Snipaste_2024-08-28_22-01-44
Snipaste_2024-08-28_22-04-09
Snipaste_2024-08-28_22-07-47

I recommend checking these display issues to ensure the chart looks clean and readable across different screen sizes.

**Comment by 20chen-7** [comment]
> Changes working well. I have one suggestion Light mode looks good. In dark mode, it is difficult to read black letters. It will be great if color can be changed to light one in dark mode. Please find images below image
> 
> image

fixed, plz check.
Screenshot 2024-08-29 at 12 55 57 PM

**Comment by 20chen-7** [comment]
> Hi, Great job! Everything looks good overall, but there are a couple of issues that appear when the screen size is between 995 and 1184 pixels: Bar Chart Labels Overlapping: The value labels on top of the bars (e.g., 122.97 and 0%) seem to overlap with each other, which can make the data hard to read. Y-Axis Alignment Issue: It looks like the bottom of the Y-axis and the bottom of the bars are not properly aligned. The "0" tick on the Y-axis is slightly offset from the base of the bars, making the chart look a bit cluttered and unclear. Snipaste_2024-08-28_22-01-44 Snipaste_2024-08-28_22-04-09 Snipaste_2024-08-28_22-07-47
> 
> I recommend checking these display issues to ensure the chart looks clean and readable across different screen sizes.

fixed, plz check.
Screenshot 2024-08-29 at 12 55 57 PM

**Review by FayeLV — APPROVED** [review]

**Review by vedantg24 — DISMISSED** [review]
Changes are working fine. It's readable in dark mode. Also, the bar chart is having no issues when the screen size is between 995 and 1184 pixels.

Screenshot 2024-08-30 at 12 12 40 AM

Screenshot 2024-08-30 at 12 14 47 AM

**Review by ashayk9 — DISMISSED** [review]
The changes are working well. It's readable in dark mode. the bar chart displays correctly when the screen size is between 995 and 1184 pixels.
Screenshot 2024-09-04 at 9 02 59 PM
Screenshot 2024-09-04 at 9 05 29 PM
Screenshot 2024-09-04 at 9 06 10 PM

**Review by crystallow1168 — APPROVED** [review]
Hi @20chen-7,

Changes looks good on light and dark mode! 

image

image

---

❗️Observation: The text and the border is overlapping when screen is 680px and below.
image

**Comment by sud306** [comment]
Hello,
I reviewed this PR and the graphs look good!
9  PR2508_1
9  PR2508_2

**Review by swaroop-2007 — APPROVED** [review]
@20chen-7 
Everything works well! I checked it with dark mode as well as light mode and in different sizes. 
Screenshot (380)
Screenshot (381)
Screenshot (382)

**Review by strallia — DISMISSED** [review]
The chart looks good and works at various browser widths. 

My only note is that in dark mode the dark gray text is hard to read against the dark blue background. I suggest adding more color contrast for readability, maybe a lighter gray.
Screenshot 2024-09-09 at 2 26 22 PM

**Review by AJAYINAVOLU — APPROVED** [review]
Changes are working as expected.

image

Thank You

**Comment by 20chen-7** [comment]
> The chart looks good and works at various browser widths.
> 
> My only note is that in dark mode the dark gray text is hard to read against the dark blue background. I suggest adding more color contrast for readability, maybe a lighter gray. Screenshot 2024-09-09 at 2 26 22 PM

left pie chart has been merged, and not related to mines. I can only change the bar chart. Plz read the screenshot in the description.

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**